rust-toolchain

```diff
@@ -1 +1 @@
-1.56.0
+1.66.0
```

Comment:
This is insufficient to produce an MSRV bump. I will do this in a separate PR.

Reply:
Also, MSRV 1.66 is a false MSRV, because it is possible to pin to `blake2b_simd 1.0.0` in order to avoid the `constant_time_eq` bumps. There's no need to raise MSRV beyond what `rand` forces.